batch_conversion.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python
+
+"""URDF files to Webots PROTO converter."""
+
+import optparse
+import datetime
+import os
+import shutil
+import json
+import math
+# from urdf2webots.importer import convert2urdf
+import urdf2webots.importer as importer
+from copy import deepcopy
+
+# Set Override = True  if you want to override ALL config settings with the
+# value in the OverrideCfg. This can be usefull for testing functionality
+# quickly, and then doing a slow optimized conversion
+Override = False
+OverrideCfg = {
+    'disableMeshOptimization': False
+}
+
+# This defines the default config. If a urdf model doesnt have a config, one
+# will created after this. If the importer.py is updated, so should this to
+# add the additional functionality-
+default_config = {
+    # outFile will get constructed with protosPath + robotName + '.proto'
+    'robotName': None,
+    'normal': False,
+    'boxCollision': True,
+    'disableMeshOptimization': True,
+    'enableMultiFile': True,
+    'staticBase': True,
+    'toolSlot': None,
+    'initRotation': "1 0 0 -1.5708"
+}
+
+
+class batchConversion():
+    def __init__(self, sourcePath, outPath):
+        self.urdf2webots_path = os.path.dirname(os.path.abspath(__file__))
+        # create a unique path for a new batch-conversion
+        today = datetime.date.today()
+        todaystr = today.isoformat()
+        self.protoTargetDir = outPath + '/protos_' + todaystr
+        n = 2
+        while os.path.exists(self.protoTargetDir):
+            while os.path.exists(self.protoTargetDir + '_Nr-' + str(n)):
+                n += 1
+            self.protoTargetDir = self.protoTargetDir + '_Nr-' + str(n)
+
+        # Find all the urdf files, and create the corresponding proto filePaths
+        urdf_root_dir = sourcePath  # 'automatic_conversion/urdf'
+        os.chdir(urdf_root_dir)
+        # Walk the tree.
+        self.urdf_files = []  # List of the full filepaths.
+        self.protosPaths = []
+        for root, directories, files in os.walk('./'):
+            for filename in files:
+                # Join the two strings in order to form the full filepath.
+                if filename.endswith(".urdf"):
+                    filepath = os.path.join(root, filename)
+                    filepath = filepath[1:]
+                    self.urdf_files.append(urdf_root_dir + filepath)
+                    self.protosPaths.append(self.protoTargetDir + os.path.dirname(filepath) + '/')
+        os.chdir(self.urdf2webots_path)
+
+        # Report dict we can print at the end
+        self.EndReportMessage = {
+            'updateConfigs': [],
+            'newConfigs': [],
+            'converted': [],
+            'failed': []
+        }
+
+    def update_and_convert(self):
+        # Make sure all configs exist and are up to date, then convert all URDF files
```

Comment:
```suggestion
        """Make sure all configs exist and are up to date, then convert all URDF files."""
```

Python function description should looks like this.